### PR TITLE
[#7] Add horizontal Categories to Home screen

### LINKED ISSUE #7: [UI] As a user, I can see the scrollable horizontal categories on the Home page
## Why

Whenever user selects the Home tab on the bottom navigation bar, the content will be shown to the user.

## Acceptance Criteria

- The list of category shown on the bottom of the "Suggested for you" card.
- User is able to scroll left/right the see all the categories on the list.

## Resources

https://www.figma.com/file/GXmbnWUEH9ZkrUluaPyedn/e-commerce?node-id=48%3A244

### PR DESCRIPTION
**Pull request by hoangnguyen92dn**
https://github.com/nimblehq/jetpack-compose-ecommerce/issues/7

## What happened 👀

Following the design on [Figma](https://www.figma.com/file/GXmbnWUEH9ZkrUluaPyedn/e-commerce?node-id=48%3A68), the Home tab shows horizontal category items
 
## Insight 📝

- Add the Category model to display the Category item.
- Add the horizontal Categories on the Home page.
 
## Proof Of Work 📹

https://user-images.githubusercontent.com/6950766/147948241-87c04b1a-8679-4baf-94d6-4982a79d7940.mp4

### CODE REVIEW

File: app/src/main/java/co/nimblehq/compose/ecommerce/ui/category/Categories.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
+package co.nimblehq.compose.ecommerce.ui.category
```
Comment (Wadeewee) [inline app/src/main/java/co/nimblehq/compose/ecommerce/ui/category/Categories.kt:1]: What do you think if we rename the class name to have a suffix with `Compose` e.g. `CategoriesCompose` like  `CategoriesActivity` or `CategoriesFragment` 🙏

File: app/src/main/java/co/nimblehq/compose/ecommerce/model/Category.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import co.nimblehq.compose.ecommerce.R
+
+@Immutable
+data class Category(
+    val id: Long,
+    val name: String,
+    @DrawableRes val image: Int
+)
+
+val categories = listOf(
```
Comment (Wadeewee) [inline app/src/main/java/co/nimblehq/compose/ecommerce/model/Category.kt:14]: Do you have a plan to remove this so far, Is it just mock data or the real data🙏

Comment (hoangnguyen92dn) [reply app/src/main/java/co/nimblehq/compose/ecommerce/model/Category.kt:14]: @Wadeewee We will use it for mocking the data. We don't have the API to do integration. I will create a fake repository to contain all the data. How do you think?

Comment (sleepylee) [reply app/src/main/java/co/nimblehq/compose/ecommerce/model/Category.kt:14]: it's ok just mock the data for now and make it more focused on the UI, @hoangnguyen92dn. Having more layers will lose the focus of the developers when reading the codebase so I advise we keep mocking it directly in the Model classes.

Comment (Wadeewee) [reply app/src/main/java/co/nimblehq/compose/ecommerce/model/Category.kt:14]: I agree to keep mock data in model class but would it be good if we leave TODO or some comment why we have to use    mock data for now and we can integrate if we have an api later something like this What do you think? @hoangnguyen92dn

File: app/src/main/java/co/nimblehq/compose/ecommerce/ui/category/Categories.kt (hunk 1 above)
Comment (hoangnguyen92dn) [reply app/src/main/java/co/nimblehq/compose/ecommerce/ui/category/Categories.kt:1]: @Wadeewee Agree 👍  I think we could add `Compose` as the convention for our team. For now, I'd prefer to go with the current approach, as you can see from [compose-samples](https://github.com/android/compose-samples), developers are not adding `Compose` to the compose functions 😄

**Comment by Wadeewee** [comment]
Rest LGTM 🚀

File: app/src/main/java/co/nimblehq/compose/ecommerce/ui/category/Categories.kt (hunk 1 above)
Comment (luongvo) [reply app/src/main/java/co/nimblehq/compose/ecommerce/ui/category/Categories.kt:1]: Agree, having Compose postfix would help us to differentiate them to model classes also. We should consider to have it in team convention.

**Review by luongvo — APPROVED** [review]
lgtm

**Pull request merged**